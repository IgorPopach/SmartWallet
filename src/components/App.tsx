import React, { useCallback } from 'react';
import { Dispatch, AnyAction, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import RegistrationForm from './Register/Form';
import { initializeSession, logIn, logOut } from '../store/session/actions';
import { AppState } from '../store';
import { User } from '../types/User';
import LogoutBtn from './LogoutBtn';
import LoginForm from './Login/Form';
import Snackbar from './Snackbar/Snackbar';
import { addMessage } from './../store/snackbar/actions';
import { AddedMessage } from './../store/snackbar/types';

interface StateProps {
    user: User | null;
}
interface DispatchProps {
    initialize: typeof initializeSession;
    onLogin: (login: string, password: string) => void;
    onLogout: () => void;
    incrementMessage: (message: AddedMessage) => void;
}

interface OwnProps {
    title: string;
    paragraph: string;
}

type Props = StateProps & DispatchProps & OwnProps;

// TODO: Temporary solution need refactoring
// tslint:disable:jsx-no-multiline-js
export const App: React.FC<Props> = ({ title, paragraph, initialize, user, onLogin, onLogout, incrementMessage }) => {
    React.useEffect(() => {
        initialize();
    }, []);

    const messageObj = {
        title,
        text: paragraph,
        type: 'success',
    };

    const addMessageCallback = useCallback(() => incrementMessage(messageObj), [messageObj]);

    return (
        <div className="container-md app">
            <Snackbar />
            <h2>{title}</h2>
            <p>{paragraph}</p>
            <button className="btn btn-danger" onClick={addMessageCallback}>
                Add this message
            </button>
            <hr />
            {!!user ? (
                <LogoutBtn onLogout={onLogout} />
            ) : (
                <>
                    <LoginForm onSubmit={onLogin} />
                    <RegistrationForm title="Register" />
                </>
            )}
        </div>
    );
};

const mapStateToProps = ({ session }: AppState) => ({
    user: session.user,
});

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
    bindActionCreators(
        {
            initialize: initializeSession,
            onLogin: logIn,
            onLogout: logOut,
            incrementMessage: addMessage,
        },
        dispatch,
    );

export default connect(mapStateToProps, mapDispatchToProps)(App);
