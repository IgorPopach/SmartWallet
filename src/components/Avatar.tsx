import React from 'react';

interface OwnProps {
    imageUrl: string | undefined;
}

const Avatar = ({ imageUrl }: OwnProps) => {
    if (!imageUrl) {
        imageUrl = 'https://gravatar.com/avatar/d560435b0ef2c7e51cbd65fd6649618c?s=400&d=mp&r=x';
    }
    return <img src={imageUrl} alt="Avatar" className="avatar" />;
};

export default Avatar;
