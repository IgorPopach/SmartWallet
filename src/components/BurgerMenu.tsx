import React from 'react';

interface Props {
    toggleAction: () => void;
    sidebarIsVisible: boolean;
}

const BurgerMenu = ({ toggleAction, sidebarIsVisible }: Props) => {
    const classes = React.useMemo(() => {
        let styles = 'burger-menu';
        if (sidebarIsVisible) {
            styles = 'burger-menu active';
        }
        return styles;
    }, [sidebarIsVisible]);

    const handleClick = React.useCallback(() => toggleAction(), []);
    return (
        <div className={classes} onClick={handleClick}>
            <span className="burger-menu-piece" />
            <span className="burger-menu-piece" />
            <span className="burger-menu-piece" />
        </div>
    );
};

export default BurgerMenu;
