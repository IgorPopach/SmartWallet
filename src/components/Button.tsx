import React, { FunctionComponent } from 'react';

import { toClasses } from '../utils';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    color: 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark' | 'link';
}

const Button: FunctionComponent<ButtonProps> = ({ className, color, children, ...shared }) => {
    const classes = React.useMemo(() => toClasses([`btn`, `btn-${color}`, className]), [color, className]);
    return (
        <button {...shared} className={classes}>
            {children}
        </button>
    );
};

Button.defaultProps = {
    type: 'button',
};

export default Button;
