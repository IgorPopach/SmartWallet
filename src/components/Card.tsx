import React from 'react';
import { IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface Props {
    icon: IconDefinition;
    title: string;
    value: string;
}

const Card = ({ icon, title, value }: Props) => {
    return (
        <div className="card card-component">
            <div className="card-body">
                <FontAwesomeIcon {...{ icon }} className="card-icon" />
                <div className="card-info">
                    <h5 className="card-title">{title}</h5>
                    <p className="card-value">{value}</p>
                </div>
            </div>
            <div className="card-footer text-muted">footer</div>
        </div>
    );
};

export default Card;
