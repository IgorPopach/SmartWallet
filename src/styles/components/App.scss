.app {
    font-family: 'Roboto', sans-serif;
    height: 100vh;
    font-size: 14px;
    position: relative;
    background-color: gray("200");
}

p {
    margin-bottom: 0.4rem;
}

.form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 250px;
    margin: -40px auto;
    height: 100vh;
    
    &-signUp {
        min-height: 420px;

    }

    h2 {
        font-size: 1.5rem;
        text-align: center;
    }
    form {
        display: flex;
        flex-direction: column;
    }
    label {
        margin-top: 10px;
        sup {
            color: red;
        }
    }
    span {
        color: red;
        font-size: 10px;
    }
    button {
        margin: 10px 0;
    }
}