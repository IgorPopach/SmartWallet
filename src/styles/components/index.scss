@import "./App.scss";
@import "./LoginForm.scss";
@import "./RegistrationForm.scss";
@import "./InputField.scss";
@import "./Alert.scss";
@import "./Snackbar.scss";
@import "./Spinner.scss";
@import "./Header.scss";
@import "./Avatar.scss";
@import "./Footer.scss";
@import "./Sidebar.scss";
@import "./BurgerMenu.scss";
@import "./ContentWrapper.scss";
@import "./AddCosts.scss";
@import "./Dropdown.scss";
@import "./Select.scss";
@import "./SelectField.scss";
@import "./Profile.scss";
@import "./UploadField.scss";
@import "./UploadView.scss";
@import "./MyCosts.scss";
@import "./Card.scss";
@import "./CostsForm.scss";
@import "./CostsTable.scss";
@import "./EditCosts.scss";
@import "./DateTimePicker.scss";
@import "./Calendar.scss";
@import "./Time.scss";
@import "./DateField.scss";