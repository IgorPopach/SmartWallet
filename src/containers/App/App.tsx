import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { History } from 'history';

import Snackbar from '../Snackbar';
import usePrevious from '../../utils/usePrevious';
import { User } from '../../types/User';
import { initializeSession, logOut } from '../../store/session/actions';
import { Spinner } from '../../components/Spinner';
import { Footer } from '../../components/Footer';
import { Header } from '../../components/Header';
import { PATH } from '../../routes/path';
import Routes from '../../routes/Routes';
import Sidebar from '../../components/Sidebar';
import ContentWrapper from '../../components/ContentWrapper';

interface OwnProps {
    history: History;
}

interface StateProps {
    isLoading: boolean;
    user: User | null;
}

interface DispatchProps {
    initialize: typeof initializeSession;
    logoutAction: typeof logOut;
}

type Props = OwnProps & StateProps & DispatchProps & RouteComponentProps;

export const App: React.FC<Props> = ({ initialize, user, isLoading, history, logoutAction }) => {
    const prevUser = usePrevious(user);

    React.useEffect(() => {
        initialize();
    }, []);

    React.useEffect(() => {
        if (prevUser === null && !!user) {
            history.push(PATH.HOME);
        }
        if (!!prevUser && user === null) {
            history.push(PATH.LOGIN);
        }
    }, [user]);

    if (isLoading) {
        return <Spinner />;
    }
    return (
        <>
            <Header {...{ logoutAction, user }} />
            <Sidebar {...{ user }} />
            <ContentWrapper>
                <Routes />
            </ContentWrapper>
            <Footer />
            <Snackbar />
        </>
    );
};
