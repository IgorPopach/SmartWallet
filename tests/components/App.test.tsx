import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import { App } from '../../src/components/App';

type Props = React.ComponentProps<typeof App>;

describe('<App />', () => {
    const props: Props = {
        title: 'Ok',
        paragraph: 'Paragraph',
        user: null,
        initialize: jest.fn(),
        onLogin: jest.fn(),
        onLogout: jest.fn(),
    };
    let wrapper: ShallowWrapper<Props>;

    beforeAll(() => {
        wrapper = shallow(<App {...props} />);
    });

    it('Should have correct layout', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
