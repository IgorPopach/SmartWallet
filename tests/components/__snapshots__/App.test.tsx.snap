// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should have correct layout 1`] = `
<div
  className="app"
>
  <Component
    user={null}
  />
  <Connect(Sidebar)
    user={null}
  />
  <ContentWrapper
    user={null}
  >
    <Routes
      user={null}
    />
  </ContentWrapper>
  <Component />
  <Connect(Snackbar) />
</div>
`;
